contracts/zksync/Ethlance.sol

```diff
@@ -189,36 +206,34 @@ contract Ethlance is ApproveAndCallFallBack, IERC721Receiver, IERC1155Receiver,
    * See spec :ethlance/job-created for the format of _ipfsData file
    * TODO: Add validation and step 2
    */
-  function _createJob(
+   // Make this normal (public) function because not all ERC20 support calling a function during
+   // token transfer
+   // The user pre-approves token transfer (call approve on their token contract)
+   // ERC721 (maybe 1155 too) (encode the function to be called in safeTransferFrom last parameter)
+
+  function createJob(
     address _creator,
-    TokenValue[] memory _offeredValues,
+    EthlanceStructs.TokenValue[] memory _offeredValues,
     JobType _jobType,
-    address[] _invitedArbiters,
+    address[] memory _invitedArbiters,
     bytes memory _ipfsData
-  ) internal {
-    address newJob = address(new MutableForwarder());
-    MutableForwarder(newJob).setTarget(jobProxyTarget);
-    Job(newJob).initialize(this, _creator, _jobType, _offeredValues, _invitedArbiters);
-    emit JobCreated(newJob, Job(newJob).version(), _jobType, _creator, _offeredValues, _invitedArbiters, _ipfsData);
+  ) public payable returns(address) {
+    address newJob = address(new MutableForwarder()); // This becomes the new proxy
+    address payable newJobPayableAddress = payable(address(uint160(newJob)));
+    uint timestamp = block.number;
+    MutableForwarder(newJobPayableAddress).setTarget(jobProxyTarget);
+
+    EthlanceStructs.transferToJob(_creator, address(this), newJobPayableAddress, _offeredValues);
+    // Job(newJobPayableAddress).initialize(this, _creator, _jobType, _offeredValues, _invitedArbiters);
+    EthlanceStructs.TokenValue[] memory toval;
+    address[] memory arbit;
+    Job(newJobPayableAddress).initialize(this, address(this), _jobType, toval, arbit);
```

Comment:
All simple values go in, the `Job#initialize` implementation is empty. Why does it revert the transaction.